src/utils/loadBrain.ts

```diff
@@ -18,25 +20,49 @@ export async function loadBrain() {
 
 /**
  * This is only exported for test
+ * This function returns a list of relative paths to directories in brain which
+ * contain a file that is not a .d.ts, .map, or .md file
  */
-export async function getBrainModules() {
-  return (await fs.readdir(ROOT)).filter(
-    (f) => !f.endsWith('.d.ts') && !f.endsWith('.map') && !f.endsWith('.md')
-  );
+export async function getBrainModules(dir: string = ROOT) {
+  // Read the directory contents
+  const files = await fs.readdir(dir, { withFileTypes: true });
+  const directories: Set<string> = new Set();
+
+  // Loop through each file or directory in the current directory
+  for (const file of files) {
+    const filePath = path.join(dir, file.name);
+
+    if (file.isDirectory()) {
+      const nestedDirs = await getBrainModules(filePath);
+      nestedDirs.forEach((nestedDir) => directories.add(nestedDir));
+    } else if (
+      !file.name.endsWith('.d.ts') &&
+      !file.name.endsWith('.map') &&
+      !file.name.endsWith('.md')
+    ) {
+      directories.add(path.relative(ROOT, dir)); // Add relative path to directory
```

Comment:
This comment doesn't add much value
```suggestion
      directories.add(path.relative(ROOT, dir));
```